### PR TITLE
Move agri2018 and agri20130 code to subfolders energy_demand, energy_source, energy_general

### PR DESCRIPTION
**Pull request by Jeniffere**
python devtool.py ready_to_rock
WARNING: there is a new pyright version available (v1.1.269).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

No configuration file found.
pyproject.toml file found at E:\LocalZero\Code\working-dir\localzero-generator-core.
Loading pyproject.toml file at E:\LocalZero\Code\working-dir\localzero-generator-core\pyproject.toml
Assuming Python platform Windows
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath E:\LocalZero\Code\working-dir\localzero-generator-core\typings is not a valid directory.
Searching for source files
Found 134 source files
0 errors, 0 warnings, 0 informations 
Completed in 8.517sec
npm notice 
npm notice New minor version of npm available! 8.3.1 -> 8.19.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.19.1
npm notice Run npm install -g npm@8.19.1 to update!
npm notice
============================================================================================================================================= test session starts ============================================================================================================================================== 
platform win32 -- Python 3.10.2, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: E:\LocalZero\Code\working-dir\localzero-generator-core
plugins: typeguard-2.13.3
collected 31 items

tests\test_devtool_commands.py ...............                                                                                                                                                                                                                                                            [ 48%] 
tests\test_end_to_end.py ...........                                                                                                                                                                                                                                                                      [ 83%]
tests\test_entries.py .                                                                                                                                                                                                                                                                                   [ 87%]
tests\test_refdata.py ....                                                                                                                                                                                                                                                                                [100%]

============================================================================================================================================= 31 passed in 19.02s ==============================================================================================================================================
Trim Trailing Whitespace.................................................Passed
Mixed line ending........................................................Passed
Check for case conflicts.................................................Passed
Check Yaml...............................................................Passed
Check for added large files..............................................Passed
Don't commit to branch...................................................Passed
black....................................................................Passed
You are ready to rock and save the climate at 8f7404ca6e13ba3853c4ee6d2a98632968587993, but don't forget to copy paste the above into your pull request
PS E:\LocalZero\Code\working-dir\localzero-generator-core>

### CODE REVIEW

File: generatorcore/agri2018/energy_demand/energyWithPercentage.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# pyright: strict
+
+from dataclasses import dataclass
+
+from ...utils import div
+
+from .energy import Energy
+
```
Comment (bgrundmann) [inline generatorcore/agri2018/energy_demand/energyWithPercentage.py:8]: I don't feel super strongly but for cases like this I think the split into one file per class is actually net negative.

I mean in this case both when I see the energy class seeing that we also have a EnergyWithPercentage class is good context and the other way around and in both cases everything would still fit on a single screen.

Or differently a module is a unit of abstraction / encapsulation.  We should use it to actually group related things together.  In my point of view I would also put the Co2eFrom* classes into one file (maybe even co2efrom.Soil) .  But the tradeoff is less clear the longer the class.

File: generatorcore/agri2018/energy_source/co2eFromEnergyUse.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# pyright: strict
+
+from dataclasses import dataclass
+
+from ..co2eEmissions import CO2eEmissions
+
+
```
Comment (bgrundmann) [inline generatorcore/agri2018/energy_source/co2eFromEnergyUse.py:7]: Same argument again.

**Review by bgrundmann — COMMENTED** [review]
I basically have only one comment which is I think you have sometimes taken the split into one file per class rule too far.

But I don't feel super strongly.  Have a think let me know if you agree or if you think I'm nuts.

**Comment by Jeniffere** [comment]
I understand your point. Although I prefer to have one class per file to keep the same level of abstraction in every file.

We talked and agreed to disagree ;)

I will merge it like this ... and we can optimize / change it later ...

Feel free to change 'my' code ;)

**Pull request merged**